crates/torii/libp2p/src/server/mod.rs

```diff
@@ -170,48 +171,60 @@ impl Relay {
                                 "Received message"
                             );
 
-                            // Check if message already exists
-                            let keys = if let Ty::Struct(s) = &message {
-                                let mut keys: Vec<starknet_ff::FieldElement> = Vec::new();
-                                for m in s.keys() {
-                                    let key = match m.serialize() {
-                                        Ok(key) => key,
-                                        Err(e) => {
-                                            info!(
-                                                target: "torii::relay::server",
-                                                error = %e,
-                                                "Failed to serialize key"
-                                            );
-                                            continue;
+                            // Message has to be a struct
+                            // that contains an identity & signature
+                            let message_struct = match message {
+                                Ty::Struct(message) => {
+                                    let mut identity: Primitive::ContractAddress;
+                                    let mut signature: Primitive::Felt252;
+                                    for member in message.keys() {
+                                        match member.name.as_str() {
+                                            "identity" => {
+                                                // check if identity is correct primitive type
+                                                if let Ty::Primitive(Primitive::ContractAddress(identity)) =
+                                                    &member.value
+                                                {
+                                                    identity = &identity.clone();
+                                                } else {
+                                                    warn!(
+                                                        target: "torii::relay::server",
+                                                        message_id = %message_id,
+                                                        "Identity is not a contract address"
+                                                    );
+                                                    continue;
+                                                }
+                                            }
+                                            "signature" => {
+                                                // check if signature is correct primitive type
+                                                if let Ty::Primitive(Primitive::Felt252(signature)) =
```

Comment:
Hmm should this be an Array<Felt252>?

Reply:
Yes, was just wondering if it would be maybe better to take in the sig_r and sig_s as two different fields as we dont support tuples right now but I guess its fine - as it wouldn't be on chain anyway.